Source/options.js

```diff
@@ -1,28 +1,49 @@
-
 function save_options() {
-  var emission = document.getElementById("emission");
-  var carbonEmission = emission.value;
-  localStorage["carbonEmission"] = carbonEmission;
+    var emission = document.getElementById("emission");
+    var unit = document.getElementById("unit");
+    var carbonEmission = emission.value;
+    var emissionUnit = unit.value;
+    switch (emissionUnit) {
+        case "kg/km":
+            carbonEmission = 1000 * carbonEmission;
+            break;
+        case "g/mi":
+            carbonEmission = 0.621371 * carbonEmission;
+            break;
+        case "kg/mi":
+            carbonEmission = 621.371 * carbonEmission;
+            break;
+        case "ltr/km":
+            carbonEmission = 2320 * carbonEmission;
+            break;
+        case "gal/km":
+            carbonEmission = 8782.155 * carbonEmission;
```

Comment:
Where did you obtain these conversion factors? On which assumptions do they depend? Could you please add some comments in the code, indicating the source that you used?

Thanks!